lightning/src/onion_message/messenger.rs

```diff
@@ -119,10 +130,21 @@ where
 	logger: L,
 	pending_messages: Mutex<HashMap<PublicKey, VecDeque<msgs::OnionMessage>>>,
 	secp_ctx: Secp256k1<secp256k1::All>,
+	message_router: MR,
 	offers_handler: OMH,
 	custom_handler: CMH,
 }
 
+/// A trait defining behavior for routing an [`OnionMessage`].
+///
+/// [`OnionMessage`]: msgs::OnionMessage
+pub trait MessageRouter {
+	/// Returns a route for sending an [`OnionMessage`] to the given [`Destination`].
```

Comment:
Make it clearer here that `find_route` should return only _intermediate_hops_  to the destination?
I would expect a `find_route` call to also include the pubkey of the final node (or intro node), but when it's used in the next commit with `send_custom_message` it's expected to only return the intermediate hops.

Also, empty vector = no path?

Reply:
> Make it clearer here that `find_route` should return only _intermediate_hops_ to the destination?

Add clarification to the docs.

> I would expect a `find_route` call to also include the pubkey of the final node (or intro node), but when it's used in the next commit with `send_custom_message` it's expected to only return the intermediate hops.

Perhaps  there's a better name than `find_route` or at very least a better name for what it is returning. Open to suggestions.

> Also, empty vector = no path?

This would be for if there is a direct connection between `sender` and `recipient`.

Reply:
Could be `get_intermediate_hops` since that's the wording used in `send_custom_message`? But no strong feelings if the docs are updated.

Reply:
@TheBlueMatt Would you be ok with that or will this make bindings difficult?

Reply:
Discussed offline and at the review club. Renamed to `find_path` taking a `Destination` by value and returning a new `OnionMessagePath` struct. Also, updated `OnionMessenger::send_custom_message` to take this struct instead of the two parameters as suggested earlier.

Regarding naming, not sure if I should rename `MessageRouter` to `OnionMessageRouter` for consistency with the new struct and `OnionMessageContents`.